src/commands/synthetics/run-tests-lib.ts

```diff
@@ -191,53 +176,86 @@ const getTestListBySearchQuery = async (
+  const testsFromTestConfigs = await getTestConfigs(config, reporter, suites)
+
+  // Grab the list of publicIds of tests to trigger from config file/env variables/CLI params, search query or test config files
+  const testIdsFromCli = config.publicIds
+  const testIdsFromSearchQuery = testsFromSearchQuery.map(({id}) => id)
+  const testIdsFromTestConfigs = testsFromTestConfigs.map(({id}) => id)
+
+  const testIdsToTrigger =
+    testIdsFromCli.length !== 0
+      ? testIdsFromCli
+      : testIdsFromSearchQuery.length !== 0
+      ? testIdsFromSearchQuery
+      : testIdsFromTestConfigs.length !== 0
+      ? testIdsFromTestConfigs
+      : []
+
+  // Create the overrides required for the list of tests to trigger
+  const triggerConfigs = testIdsToTrigger.map((id) => {
+    const testFromSearchQuery = testsFromSearchQuery.find((test) => test.id === id)
+    const testFromTestConfigs = testsFromTestConfigs.find((test) => test.id === id)
+
+    return {
+      id,
+      ...testFromSearchQuery,
+      ...testFromTestConfigs,
+      config: {
+        ...defaultTestConfigOverrides,
+        ...testConfigOverridesFromEnv,
+        ...testFromTestConfigs?.config,
+      },
     }
+  })
 
-    return testsToTriggerBySearchQuery
+  return triggerConfigs
+}
+
+const getTestsFromSearchQuery = async (api: APIHelper, config: RunTestsCommandConfig, reporter: MainReporter) => {
+  const testsToTriggerBySearchQuery = await getTestListBySearchQuery(api, config.global, config.testSearchQuery || '')
+
+  if (testsToTriggerBySearchQuery.length > MAX_TESTS_TO_TRIGGER) {
+    reporter.error(
+      `More than ${MAX_TESTS_TO_TRIGGER} tests returned by search query, only the first ${MAX_TESTS_TO_TRIGGER} will be fetched.\n`
+    )
   }
 
+  return testsToTriggerBySearchQuery
+}
+
+const getTestConfigs = async (
+  config: RunTestsCommandConfig,
+  reporter: MainReporter,
+  suites: Suite[] = []
+): Promise<TriggerConfig[]> => {
   const suitesFromFiles = (await Promise.all(config.files.map((glob: string) => getSuites(glob, reporter))))
     .reduce((acc, val) => acc.concat(val), [])
     .filter((suite) => !!suite.content.tests)
 
   suites.push(...suitesFromFiles)
 
-  const configFromEnvironment = config.locations?.length ? {locations: config.locations} : {}
-
-  const overrideTestConfig = (test: TriggerConfig): UserConfigOverride =>
-    // {} < global < ENV < test file
-    ({
-      ...config.global,
-      ...configFromEnvironment,
-      ...test.config,
-    })
-
-  const testsToTrigger = suites
+  const TestConfigs = suites
     .map((suite) =>
       suite.content.tests.map((test) => ({
-        config: overrideTestConfig(test),
+        config: test.config,
         id: normalizePublicId(test.id) ?? '',
         suite: suite.name,
       }))
     )
     .reduce((acc, suiteTests) => acc.concat(suiteTests), [])
 
-  return testsToTrigger
+  return TestConfigs
```

Comment:
In typescript, we use capitalised name only for types and classes (which are a kind of type):
```suggestion
  return testConfigs
```

Reply:
ah yup must have mistyped it (ba-dum-ts) :D